OutOfSchool/OutOfSchool.WebApi/Services/PhotoStorage/PhotoStorage.cs

```diff
@@ -0,0 +1,350 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Localization;
+using OutOfSchool.Services.Enums;
+using OutOfSchool.Services.Models;
+using OutOfSchool.Services.Repository;
+using OutOfSchool.WebApi.Extensions;
+using OutOfSchool.WebApi.Models;
+using Serilog;
+
+namespace OutOfSchool.WebApi.Services.PhotoStorage
+{
+    public class PhotoStorage : IPhotoStorage
+    {
+        private readonly IEntityRepository<Photo> repository;
+        private readonly ILogger logger;
+        private readonly IStringLocalizer<SharedResource> localizer;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PhotoStorage"/> class.
+        /// </summary>
+        /// <param name="repository">Repository.</param>
+        /// <param name="logger">Logger.</param>
+        /// <param name="localizer">Localizer.</param>
+        public PhotoStorage(IEntityRepository<Photo> repository, ILogger logger, IStringLocalizer<SharedResource> localizer)
+        {
+            this.repository = repository;
+            this.logger = logger;
+            this.localizer = localizer;
+        }
+
+        public static string FilePath { get; set; }
+
+        /// <inheritdoc/>
+        public async Task<PhotoDto> AddFile(PhotoDto photo, string fileName)
+        {
+            try
+            {
+                logger.Information("Process of creating photo started.");
+
+                if (photo is null)
+                {
+                    throw new ArgumentNullException(localizer["Photo can not be null!."]);
+                }
+
+                if (fileName is null)
+                {
+                    throw new ArgumentNullException(localizer["File name can not be null!."]);
+                }
+
+                var dirPath = Path.GetDirectoryName(FilePath);
+
+                if (!Directory.Exists(dirPath))
+                {
+                    Directory.CreateDirectory(dirPath);
+                }
+
+                var filePath = $"{FilePath}\\{fileName.TrimEnd()}";
+
+                using (var fileStream = File.Open(filePath, FileMode.OpenOrCreate))
+                {
+                    await fileStream.WriteAsync(photo.Photo, 0, photo.Photo.Length).ConfigureAwait(false);
+                }
+
+                photo.Path = filePath;
+
+                var photoInfo = await repository.Create(photo.ToDomain()).ConfigureAwait(false);
+
+                logger.Information($"Photo with Id = {photoInfo?.Id} created successfully.");
+
+                return photoInfo.ToModel();
+            }
+            catch (ArgumentException ex)
```

Comment:
Since the handling logic is the same, we can avoid duplication by using Exception filters.

```suggestion
            catch (Exception ex) when (ex is ArgumentException || ex is IOException)
```